File: resources/exercises/sort/dutchflag/DutchFlagAlgoEntity.c
//RemoteFlag

#define BLUE  0
#define WHITE  1
#define RED  2


/* BEGIN TEMPLATE */
void solve() {
	/* BEGIN SOLUTION */
	int afterBlue=0;
	int beforeWhite=getSize()-1;
	int beforeRed=getSize()-1;
	while (afterBlue <= beforeWhite) {
		switch (getColor(afterBlue)) {
		case BLUE:
			afterBlue++;
			break;
		case WHITE:
			swap(afterBlue, beforeWhite);
			beforeWhite --;
			break;
		case RED:
			swap(afterBlue, beforeWhite);
			swap(beforeRed, beforeWhite);
			beforeWhite--;
			beforeRed--;
		}
	}
	assertSorted();
	/* END SOLUTION */
}
/* END TEMPLATE */

void run(){
	solve();
}

File: resources/exercises/sort/dutchflag/DutchFlagAlgoEntity.py
# BEGIN SOLUTION 
afterBlue=0;
beforeWhite=getSize()-1;
beforeRed=getSize()-1;
while afterBlue <= beforeWhite:
	
	if (getColor(afterBlue) == BLUE):
		afterBlue += 1
	elif (getColor(afterBlue) == WHITE):
		swap(afterBlue, beforeWhite);
		beforeWhite -= 1
	else:
		swap(afterBlue, beforeWhite);
		swap(beforeRed, beforeWhite);
		beforeWhite -= 1
		beforeRed -= 1
# END SOLUTION
